### PR TITLE
feat: adding consumer support for new traces/spans api

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
As part of supporting `fetch_total` as request in spans/traces API, this PR makes the corresponding changes at client side.

Ref PR: https://github.com/hypertrace/gateway-service/pull/169

### CODE REVIEW

File: hypertrace-core-graphql-platform/build.gradle.kts [hunk 1]
```diff
@@ -16,7 +16,7 @@ dependencies {
     api("org.hypertrace.core.grpcutils:grpc-context-utils:0.12.1")
     api("org.hypertrace.core.grpcutils:grpc-client-utils:0.12.1")
     api("org.hypertrace.core.grpcutils:grpc-client-rx-utils:0.12.1")
-    api("org.hypertrace.gateway.service:gateway-service-api:0.2.25")
+    api("org.hypertrace.gateway.service:gateway-service-api:0.3.0-SNAPSHOT")
```
Comment (kotharironak) [inline hypertrace-core-graphql-platform/build.gradle.kts:19]: Once I merged the gateway-service PR and cut the release, will update the appropriate version here.

File: hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/request/DefaultSpanRequestBuilder.java [hunk 2]
```diff
@@ -54,12 +68,13 @@ public Single<SpanRequest> build(
...
       List<AttributeExpression> spanAttributeExpressions,
       List<AttributeExpression> logAttributeExpressions) {
+
     return zip(
         resultSetRequestBuilder.build(
             context, HypertraceCoreAttributeScopeString.SPAN, arguments, spanAttributeExpressions),
         logEventAttributeRequestBuilder.buildAttributeRequest(context, logAttributeExpressions),
         (resultSetRequest, logEventAttributeRequest) ->
-            new DefaultSpanRequest(context, resultSetRequest, logEventAttributeRequest));
+            new DefaultSpanRequest(context, resultSetRequest, logEventAttributeRequest, false));
```
Comment (kotharironak) [inline hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/request/DefaultSpanRequestBuilder.java:77]: This is used via ExportSpans API, and `ExportSpanResult` doesn't extends `ResultSet`. It doesn't support total.
Ref: https://github.com/hypertrace/hypertrace-core-graphql/blob/main/hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/schema/ExportSpanResult.java#L15C1-L15C1

Comment (satish-mittal) [reply hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/request/DefaultSpanRequestBuilder.java:77]: > This is used via ExportSpans API, and ExportSpanResult doesn't extends ResultSet. It doesn't support total.

We can add it as a comment to explain why fetchTotal is set to false here.

**Review by skjindal93 — COMMENTED** [review]
looks good (apart from @satish-mittal's review comment)

**Review by satish-mittal — APPROVED** [review]

**Review by skjindal93 — APPROVED** [review]

**Pull request merged**